RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducerFromCUDA.cc

```diff
@@ -1,44 +1,215 @@
-#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "HeterogeneousCore/CUDACore/interface/GPUCuda.h"
+#include "HeterogeneousCore/CUDAServices/interface/CUDAService.h"
+#include "HeterogeneousCore/Producer/interface/HeterogeneousEDProducer.h"
+#include "RecoLocalTracker/SiPixelRecHits/plugins/siPixelRecHitsHeterogeneousProduct.h"
+
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
 #include "RecoTracker/TkHitPairs/interface/RegionsSeedingHitSets.h"
 
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+
+#include "RecoPixelVertexing/PixelTrackFitting/interface/PixelTrackBuilder.h"
+#include "RecoPixelVertexing/PixelTrackFitting/interface/PixelTrackCleaner.h"
+
+// track stuff
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/TrackExtra.h"
+#include "DataFormats/Common/interface/OrphanHandle.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "RecoPixelVertexing/PixelTriplets/plugins/pixelTuplesHeterogeneousProduct.h"
+#include "storeTracks.h"
+
+
 /**
  * This class will eventually be the one creating the reco::Track
  * objects from the output of GPU CA. Now it is just to produce
  * something persistable.
  */
-class PixelTrackProducerFromCUDA: public edm::global::EDProducer<> {
+class PixelTrackProducerFromCUDA: public HeterogeneousEDProducer<heterogeneous::HeterogeneousDevices<
+          heterogeneous::GPUCuda, heterogeneous::CPU>> {
```

Comment:
There are no GPU operations added, right? Then the class could (should) stay as `edm::global::EDProducer`.

Reply:
I understood that a global::EDProducer cannot consume Heterogeneous Product (yet)

Reply:
Ok, I see. You could do
```cpp
edm::Handle<HeterogeneousProduct> gh;
iEvent.getByToken(gpuToken_, gh); // edm::Event
auto const& tuples = gh->get<Input>.getProduct<HeterogeneousDevice::kCPU>();
```
from a normal EDProducer, but I leave it to your discretion which one feels simpler for now.

Reply:
Ok, I will try next iteration.
On the other hand I would really like to see this PR merged, just to mark a new reference.
I have already in the pipeline the mods to speed up the fit by a factor 4 and it maybe better not to mix with this

Reply:
I agree, it would be good to get this one merged :) It's not a big deal anyway, and likely it's best to let it be as it is until the migration-to-#100 reaches this point.